extensions/mssql/src/hdfs/aclEntry.ts

```diff
@@ -135,12 +137,42 @@ export class AclEntry {
 	 * Example strings :
 	 *		user:bob:rwx
 	 *		default:user:bob:rwx
-	 * 		user::r-x
-	 * 		default:group::r--
+	 *		user::r-x
+	 *		default:group::r--
 	 */
-	toString(): string {
-		return `${this.scope === AclEntryScope.default ? 'default:' : ''}${this.type}:${this.name}:${this.permission.toString()}`;
+	toAclString(): string {
+		return `${this.scope === AclEntryScope.default ? 'default:' : ''}${getAclEntryType(this.type)}:${this.name}:${this.permission.toString()}`;
 	}
+
+	/**
+	 * Checks whether this and the specified AclEntry are equal. Two entries are considered equal
+	 * if their scope, type and name are equal.
+	 * @param other The other entry to compare against
+	 */
+	public isEqual(other: AclEntry): boolean {
+		if (!other) {
+			return false;
+		}
+		return this.scope === other.scope &&
+			this.type === other.type &&
+			this.name === other.name;
+	}
+}
+
+/**
+ * Maps the possible entry types into their corresponding values for using in an ACL string
+ * @param type The type to convert
+ */
+function getAclEntryType(type: AclEntryType | AclPermissionType) {
```

Comment:
Return type for function might be helpful here

Reply:
Whoops! I'd love to enable the tslint rule for catching that but trying that came up with a couple thousand hits so I think I'll hold off on that for now 😄